src/explore-education-statistics-common/src/modules/table-tool/utils/getDefaultTableHeadersConfig.ts

```diff
@@ -90,19 +94,34 @@ function getFixedTimePeriodAndIndicatorTableHeadersConfig(
  * to create the best possible table for them.
  */
 export default function getDefaultTableHeaderConfig(
-  fullTableMeta: FullTableMeta,
+  fullTable: FullTable,
 ): TableHeadersConfig {
+  const {
+    indicators,
+    filters,
+    locations,
+    timePeriodRange,
+  } = fullTable.subjectMeta;
+  const { results } = fullTable;
+
+  // When terms are selected the time period range can include ones not displayed in the table,
+  // filter these out to ensure the reorder headings list is correct
+  const filteredTimePeriodRange = timePeriodRange.filter(period =>
+    results.find(result => result.timePeriod === period.value) ? period : null,
```

Comment:
I think you shouldn't need the `? period : null` part, as `filter()` wants a boolean returned and `results.find(...)` is already doing that for you here.

So I think it can be rewritten:

```
const filteredTimePeriodRange = timePeriodRange.filter(period =>
    results.find(result => result.timePeriod === period.value));
```